File: 200-300/258. 各位相加.py
# 1
class Solution:
    def addDigits(self, num: int) -> int:
        if num < 10: return num
        while (num > 9 ):
            ans = 0 
            while (num !=0):
                ans += num % 10
                num //= 10
            num = ans
        return ans 
#2 数学法：数根
class Solution:
    def addDigits(self, num: int) -> int:
        return (num -1) % 9 + 1 if num else num

File: 200-300/258. 各位相加.go
// 1
func addDigits(num int) int {
    for num > 9 {
        ans:= 0
        for ; num!=0 ; num /= 10 {
            ans+= num%10
        }
        num = ans
    }
    return num

}
// 2 
func addDigits(num int) int {
    if num == 0 {
        return 0
    }
    return (num -1) % 9 + 1

}